File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.go
func numSteps(s string) int {
	n := len(s)
	ans := n - 1
	carry := 0

	for i := n - 1; i > 0; i-- {
		bit := int(s[i]-'0') + carry

		if bit == 1 {
			ans++
			carry = 1
		}
	}

	return ans + carry
}

File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.kt
class Solution {
    fun numSteps(s: String): Int {
        val n = s.length
        var ans = n - 1
        var carry = 0

        for (i in n - 1 downTo 1) {
            val bit = (s[i] - '0') + carry

            if (bit == 1) {
                ans++
                carry = 1
            }
        }

        return ans + carry
    }
}